addons/udes_stock/models/stock_move.py

```diff
@@ -37,7 +37,8 @@ def _action_cancel(self):
         - Include a hook to propagate cancellations when the picking
           type is configured with u_propagate_cancel set to True
         """
-        self.move_line_ids.write({"qty_done": 0})
+        if self.move_line_ids:  # Merging moves may assign move lines to another move and cancel this, skip those
```

Comment:
This change was required to avoid an infinite loop; and it also avoids pointless chain of calls.